synapse/rest/client/report_event.py

```diff
@@ -61,6 +62,14 @@ async def on_POST(
                 Codes.BAD_JSON,
             )
 
+        event = await self._event_handler.get_event(
+            requester.user, room_id, event_id, show_redacted=False
+        )
+        if event is None:
+            raise NotFoundError(
+                "Unable to report event: it does not exist or you aren't able to see it."
+            )
```

Comment:
This isn't really part of the spec, but I guess a 404 is reasonable?

Reply:
yeah, best I can think of short of speccing something